receiver/faroreceiver/receiver.go

```diff
@@ -0,0 +1,240 @@
+	if req.Method == http.MethodOptions {
+		resp.WriteHeader(http.StatusOK)
+		return
+	}
+
+	if req.Method != http.MethodPost {
+		errorHandler(resp, req, "only POST method is supported", http.StatusMethodNotAllowed)
+		return
+	}
+
+	if req.Header.Get("Content-Type") != "application/json" {
+		errorHandler(resp, req, "invalid Content-Type", http.StatusUnsupportedMediaType)
+		return
+	}
+
+	body, err := io.ReadAll(req.Body)
+	if err != nil {
+		errorHandler(resp, req, fmt.Sprintf("failed to read request body: %v", err), http.StatusBadRequest)
+		return
+	}
+	defer req.Body.Close()
+
+	var payload faro.Payload
+	if err := json.Unmarshal(body, &payload); err != nil {
+		errorHandler(resp, req, fmt.Sprintf("failed to parse Faro payload: %v", err), http.StatusBadRequest)
+		return
+	}
+
+	var errors []string
+
+	if r.nextTraces != nil {
+		traces, err := farotranslator.TranslateToTraces(req.Context(), payload)
+		if err == nil {
+			if traces != nil {
+				err = r.nextTraces.ConsumeTraces(req.Context(), *traces)
+				if err != nil {
+					errors = append(errors, fmt.Sprintf("failed to push traces: %v", err))
+				}
+			} else {
+				r.settings.Logger.Debug("Faro traces are nil, skipping")
+			}
+		} else {
+			errors = append(errors, fmt.Sprintf("failed to convert Faro traces: %v", err))
+		}
+	} else {
+		r.settings.Logger.Debug("Traces consumer not registered, skipping")
+	}
+
+	if r.nextLogs != nil {
+		logs, err := farotranslator.TranslateToLogs(req.Context(), payload)
+		if err == nil {
+			if logs != nil {
+				err = r.nextLogs.ConsumeLogs(req.Context(), *logs)
+				if err != nil {
+					errors = append(errors, fmt.Sprintf("failed to push logs: %v", err))
+				}
+			} else {
+				r.settings.Logger.Debug("Faro logs are nil, skipping")
+			}
+		} else {
+			errors = append(errors, fmt.Sprintf("failed to convert Faro logs: %v", err))
+		}
+	} else {
+		r.settings.Logger.Debug("Logs consumer not registered, skipping")
+	}
+
+	if len(errors) > 0 {
+		r.settings.Logger.Error("Failed to process Faro payload", zap.Any("payload", payload), zap.Any("errors", errors))
+		errorHandler(resp, req, strings.Join(errors, "\n"), http.StatusInternalServerError)
+		return
+	}
+
+	resp.WriteHeader(http.StatusOK)
+}
+
+func errorHandler(w http.ResponseWriter, _ *http.Request, errMsg string, statusCode int) {
+	s := httpHelper.NewStatusFromMsgAndHTTPCode(errMsg, statusCode)
+	writeStatusResponse(w, jsEncoder, statusCode, s.Proto())
+}
+
+func writeStatusResponse(w http.ResponseWriter, enc encoder, statusCode int, rsp *spb.Status) {
+	msg, err := enc.marshalStatus(rsp)
+	if err != nil {
+		writeResponse(w, enc.contentType(), 500, []byte(err.Error()))
+		return
+	}
+	writeResponse(w, enc.contentType(), statusCode, msg)
+}
+
+func writeResponse(w http.ResponseWriter, contentType string, statusCode int, msg []byte) {
+	w.Header().Set("Content-Type", contentType)
+	w.WriteHeader(statusCode)
+	// Nothing we can do with the error if we cannot write to the response.
```

Comment:
We can (and probably should) log an error if we're unable to respond.

Reply:
Agreed logging out